server/container/guice/guice-common/src/main/java/org/apache/james/utils/InMemoryMailRepositoryStore.java

```diff
@@ -106,7 +106,7 @@ public void init() throws Exception {
         return destinationToRepositoryAssociations
                 .keySet()
                 .stream()
-                .filter((MailRepositoryUrl key) -> key.getPath().equals(path.asString()))
+                .filter((MailRepositoryUrl key) -> key.getPath().asString().equals(path.asString()))
```

Comment:
Can't we rely on path equals?

Reply:
We can rely on `equals`, but we need `MailRepositoryUrl` to fetch them from the `Map`s